packages/web-components/src/dialog/dialog.ts

```diff
@@ -157,244 +49,35 @@ export class Dialog extends FASTElement {
    */
   public show(): void {
     Updates.enqueue(() => {
-      if (this.modalType === DialogModalType.alert || this.modalType === DialogModalType.modal) {
+      if (this.type === DialogType.alert || this.type === DialogType.modal) {
         this.dialog.showModal();
-        this.open = true;
-        this.updateTrapFocus(true);
-      } else if (this.modalType === DialogModalType.nonModal) {
+      } else if (this.type === DialogType.nonModal) {
         this.dialog.show();
-        this.open = true;
       }
-      this.onOpenChangeEvent();
+      this.emitOpenChange();
     });
   }
 
   /**
    * @public
    * Method to hide the dialog
-   * @param dismissed - Indicates whether the dialog was dismissed
    */
-  public hide(dismissed: boolean = false): void {
+  public hide(): void {
     this.dialog.close();
-    this.open = false;
-    this.onOpenChangeEvent(dismissed);
+    this.emitOpenChange();
   }
 
   /**
    * @public
-   * Method to dismiss the dialog
-   */
-  public dismiss(): void {
-    if (this.modalType === DialogModalType.alert) {
-      return;
-    }
-    this.hide(true);
-  }
-
-  /**
-   * @public
-   * Handles click events on the dialog
+   * Handles click events on the dialog overlay for light-dismiss
    * @param event - The click event
    * @returns boolean
    */
   public handleClick(event: Event): boolean {
```

Comment:
nit: I usually prefer to use the format `${event}Handler` for event handler functions, so this could be `clickHandler` instead of `handleClick`.